frontend/model/contracts/chatroom.js

```diff
@@ -136,6 +141,80 @@ function emitMessageEvent ({ contractID, hash }: {
   sbp('okTurtles.events/emit', `${CHATROOM_MESSAGE_ACTION}-${contractID}`, { hash })
 }
 
+export function makeMentionFromUsername (username: string): {
+  me: string, all: string
+} {
+  return {
+    me: `@${username}`,
+    all: '@here'
+  }
+}
+
+function addMentioning ({ contractID, messageId, datetime, text, username, chatRoomName }: {
+  contractID: string,
+  messageId: string,
+  datetime: string,
+  text: string,
+  username: string,
+  chatRoomName: string
+}): void {
+  /**
+   * If 'READY_TO_JOIN_CHATROOM' is false, it means not syncing chatroom
+  */
+  if (sbp('okTurtles.data/get', 'READY_TO_JOIN_CHATROOM')) {
+    return
+  }
+
+  sbp('state/vuex/commit', 'addChatRoomUnreadMentioning', {
+    chatRoomId: contractID,
+    messageId,
+    createdDate: new Date(datetime).getTime()
+  })
+
+  const rootGetters = sbp('state/vuex/getters')
+  // Commented to make notification for mentions
+  // const isExist = rootGetters.currentGroupNotifications.find(n =>
+  //   n.type === 'MENTION_ADDED' && n.linkTo === `/group-chat/${contractID}`
+  // )
+  // if (!isExist) {
+  //   const rootState = sbp('state/vuex/state')
+  //   const groupID = rootGetters.groupIdFromChatRoomId(contractID)
+  //   sbp('gi.notifications/emit', 'MENTION_ADDED', {
+  //     groupID: groupID,
+  //     chatRoomId: contractID,
+  //     username: rootState.loggedIn.username
+  //   })
+  // }
+
+  const groupID = rootGetters.groupIdFromChatRoomId(contractID)
+  makeNotification({
+    title: `# ${chatRoomName}`,
+    body: text,
+    icon: rootGetters.globalProfile2(groupID, username).picture
+  })
+
+  sbp('okTurtles.events/emit', MESSAGE_RECEIVE, {})
```

Comment:
It is not necessary to pass in `{}` if that parameter isn't used. Can be removed.